src/char/src/ccharisc.cpp

```diff
@@ -144,6 +192,10 @@ void CCharISC::OnConnected() {
 bool CCharISC::OnShutdown() {
   logger_->trace("CCharISC::OnShutdown()");
   bool result = true;
+  if(state_ != eSTATE::REGISTERED) {
+    logger_->warn("ISC {} is attempting to shutdown before registering.", get_id());
```

Comment:
Do we care that much if the server isn't registered before shutting down the connection?

Reply:
Yes, we don't want to give an attacker the ability to send the login server server_disconnected packets when they aren't a valid server